CRM/Mailing/BAO/Mailing.php

```diff
@@ -231,6 +231,7 @@ public static function getRecipients($mailingID) {
       $criteria = array(
         'is_opt_out' => CRM_Utils_SQL_Select::fragment()->where("$contact.is_opt_out = 0"),
         'is_deceased' => CRM_Utils_SQL_Select::fragment()->where("$contact.is_deceased <> 1"),
+        'do_not_sms' => CRM_Utils_SQL_Select::fragment()->where("$contact.do_not_sms = 0"),
         'location_filter' => CRM_Utils_SQL_Select::fragment()->where("$entityTable.phone_type_id = " . CRM_Core_PseudoConstant::getKey('CRM_Core_DAO_Phone', 'phone_type_id', 'Mobile')),
```

Comment:
Nice work! I don't want to hold this PR back (and this could be a separate issue) but since it's related to this change I'll mention it:  If someone opts out of email (is_opt_out), should they really opt out of SMS too? The UI very clearly says no bulk emails, so I would be in favor of removing that query item for sending mass SMS so people do get an SMS message even if they have opted out of bulk email.

Reply:
Rather than slow down this PR, I just opened a separate issue:

https://lab.civicrm.org/dev/core/issues/318

Let's merge this PR :).